src/components/data/DataTable.js

```diff
@@ -120,8 +120,9 @@ const DataTable = props => {
     Utils.withBusyState(setLoading),
     withErrorReporting('Error loading entities')
   )(async () => {
-    const results = await Ajax(signal).Workspaces.workspace(namespace, name).entitiesOfType(entityType)
-    setSelected(entityMap(results))
+    const params = _.merge({ pageSize: filteredCount }, activeTextFilter ? { filterTerms: activeTextFilter } : {})
```

Comment:
is there a better way to do this? I never want to send an empty filterTerms query parameter, i.e. `?filterTerms=&pageSize=10` should never happen.

Reply:
When testing the page I am able to send a request with a pageSize of `0`, which results in this error:

![image](https://user-images.githubusercontent.com/50371603/136622540-611dedb8-19f0-4f29-9f49-68b14862adce.png)

I believe this will alleviate the error by always removing empty terms:
```suggestion
    const params = _.pickBy(_.identity(_.trim), { pageSize: filteredCount, filterTerms: activeTextFilter })
```

There is an additional bug on the page that is outside of the scope of this change. If you type some spaces in the search field and hit enter, it will result in a server error. We need to trim the input in the search box, but it appears there is a bug in the underlying component that is preventing this.

I created a bug to track this [here ](https://broadworkbench.atlassian.net/jira/software/c/projects/TUB/boards/110?modal=detail&selectedIssue=TUB-18)as I am not sure where else to track core component bugs.

Reply:
* good find on the behavior with zero results! I've removed the menu item in that case, as it doesn't make sense.
* thank you for the suggestion of `_.pickBy`! In your suggested code, does it really need the extra `_.identity` step? I've added a commit without identity, how does that look?
* I've added a few tweaks to dealing with whitespace-only in the text filter. I doubt this is a comprehensive fix but it seems to help; I'll leave any further fixes as out of scope.

Back to you for re-review, thank you!